lib/valkyrie/persistence/solr/repository.rb

```diff
@@ -39,20 +45,31 @@ def add_documents(documents)
     end
     # rubocop:enable Style/IfUnlessModifier
 
+    # Deletes a Solr Document using the ID
+    # @return [Array<Valkyrie::Resource>] resources which have been deleted from Solr
     def delete
       connection.delete_by_id resources.map { |resource| resource.id.to_s }, params: COMMIT_PARAMS
       resources
     end
 
+    # Given a Valkyrie Resource, generate the Hash for the Solr Document
+    # @param [Valkyrie::Resource] resource
+    # @return [Hash]
     def solr_document(resource)
       resource_factory.from_resource(resource: resource).to_h
     end
 
+    # Given a new Valkyrie Resource, generate a random UUID and assign it to the Resource
+    # @param [Valkyrie::Resource] resource
+    # @param [String] the UUID for the new resource
     def generate_id(resource)
       Valkyrie.logger.warn "The Solr adapter is not meant to persist new resources, but is now generating an ID."
       resource.id = SecureRandom.uuid
     end
 
+    # If a 409 error response is encountered when attempting to commit updates to Solr, raise a StaleObjectError
+    # @see https://lucene.apache.org/solr/guide/7_2/updating-parts-of-documents.html#optimistic-concurrency
```

Comment:
Same as above, `https://lucene.apache.org/solr/guide/updating-parts-of-documents.html#optimistic-concurrency`?  And this one uses `7_2` instead of `7_4`, so we should probably change that if we want to keep the URLs with version numbers.